cpp/memilio/mobility/metapopulation_mobility_instant.h

```diff
@@ -302,8 +391,44 @@ class MigrationEdge
     bool m_return_migrated;
     double m_t_last_dynamic_npi_check               = -std::numeric_limits<double>::infinity();
     std::pair<double, SimulationTime> m_dynamic_npi = {-std::numeric_limits<double>::max(), SimulationTime(0)};
+    std::vector<std::vector<size_t>> m_save_indices; // groups of indices from compartments to save
+    TimeSeries<double> m_mobility_results; // save results from edges + entry for the total number of commuters
+
+    /**
+     * Computes a condensed version of m_migrated and puts it in m_mobility_results.
+     * m_mobility_results then only contains commuters with infection states InfectedNoSymptoms and InfectedSymptoms.
+     * Additionally, the total number of commuters is stored in the last entry of m_mobility_results.
+     * @param[in] t current time
+     */
+    void condense_m_mobility(const double t);
```

Comment:
I would much prefer if this mobility result could be computed outside of this class in a post-processing. As far as I can see, that *is* exactly how m_mobility_results is currently being used. Or is it (supposed to be) used anywhere during a simulation? If not, I propose that we use a free function (e.g. "condense_mobility_result" or "accumulate_compartments") that takes a TimeSeries (e.g. m_migrated) as well as those indices and returns a condensed TimeSeries, instead of the class member. This solution is a lot more flexible, e.g. it allows multiple "condensations" in one simulation, and the condense function could be reused for non-mobility results.

Reply:
As already discussed, post-processing is not possible here, as we reset the results to the m_migrated after each day.
Therefore, the current solution is flexible within the scope of the possibilities. I have tried the idea of integrating a history object like in the advance function as already used in the ABM, but this requires a lot of changes, so I would leave it at the current version